CMakeLists.txt

```
@@ -0,0 +1,31 @@
+set(WORMHOLE_WILLIAM_LIB
+    ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libwormhole_william.so)
+
+if(${CMAKE_HOST_SYSTEM_NAME} STREQUAL "Windows")
+  set(EXPORT set)
+else()
+  set(EXPORT export)
+endif()
+
+add_custom_target(
+  BUILD_GO_LIBRARY ALL
+  COMMENT Building go static library
+  COMMAND ${EXPORT} CGO_ENABLED=1&&
+  	  ${EXPORT} GOARCH=${GOARCH}&&
+	  ${EXPORT} GOOS=${GOOS}&&
+	  ${EXPORT} CC_FOR_TARGET=${CC_FOR_TARGET}&&
+	  ${EXPORT} CC=${CC_FOR_TARGET}&&
+	  go build -buildmode=c-shared -o ${WORMHOLE_WILLIAM_LIB} ./c/client.c.go
```

It looks like we're still only building the one file rather than the entire package.
```suggestion
	  go build -buildmode=c-shared -o ${WORMHOLE_WILLIAM_LIB} ./c
```

---

I reverted this actually, tried again and it doesn't compile, so I brought it back and did a test to ensure the dependent code is there and working :)

---

This looks like go figuring out what code needs to be built rather than us telling it the exact files.